corehq/apps/cloudcare/views.py

```diff
@@ -602,12 +603,15 @@ def _fail(error):
     restore_as_user, set_cookie = FormplayerMain.get_restore_as_user(request, domain)
     force_login_as = not restore_as_user.is_commcare_user()
     if force_login_as and not can_use_restore_as(request):
-        _fail(_("This user cannot access this link."))
+        return _fail(_("This user cannot access this link."))
 
     cloudcare_state = json.dumps({
         "appId": build._id,
         "endpointId": endpoint_id,
-        "endpointArgs": request.GET,
+        "endpointArgs": {
+            six.moves.urllib.parse.quote_plus(key): six.moves.urllib.parse.quote_plus(value)
```

Comment:
I don't think there's any reason to use the six version:
```suggestion
            urllib.parse.quote_plus(key): urllib.parse.quote_plus(value)
```

Reply:
👍  ad41e771956